docs/user/registering-microshift-devices-acm.md

```diff
@@ -70,94 +71,29 @@ spec:
-        repository: acm-registration
-        suffix: /agent-registration/manifests/{{ device.metadata.name }}
-    ```
-
-    This API call to ACM's agent registration endpoint will retrieve a set of Kubernetes manifests to deploy the Klusterlet. Once we have both the Klusterlet CRD and deployment manifests, we will use Flight Control hooks to apply them.
-
-    ```console
-    - name: pull-secret
-      inline:
-          - path: "/etc/crio/openshift-pull-secret"
-            content: "{\"auths\":{...}}"
-    ```
-
-    This adds your OpenShift pull secret to the device.  You can download your pull secret [here](https://cloud.redhat.com/openshift/install/pull-secret).  If you do not wish to have your pull secret available via the Flight Control API, you may store it in a private Git repository or a Kubernetes Secret, and change the fleet specification accordingly.
-
-1. **Hooks**: Once the configuration is fetched, the hooks apply the Kubernetes manifests to the device:
-
-```console
-hooks:
-  afterUpdating:
-    - path: "/var/local/acm-import/crd.yaml"
-      onFile: [Create]
-      actions:
-        - executable:
-            run: "kubectl apply -f /var/local/acm-import/crd.yaml"
-            envVars: ["KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig"]
-            workDir: "/usr/bin/"
-    - path: "/var/local/import.yaml"
-      onFile: [Create]
-      actions:
-        - executable:
-            run: "kubectl apply -f /var/local/acm-import/import.yaml"
-            envVars: ["KUBECONFIG=/var/lib/microshift/resources/kubeadmin/kubeconfig"]
-            workDir: "/usr/bin/"
+        - path: "/etc/crio/openshift-pull-secret"
+          content: "{\"auths\":{...}}"
+      - name: apply-acm-manifests
+        inline:
+        - path: "/etc/flightctl/hooks.d/afterupdating/50-acm-registration.yaml"
+          content: |
+            - if:
+              - path: /var/local/acm-import/crd.yaml
+                op: [created]
+              run: kubectl apply -f /var/local/acm-import/crd.yaml
+              envVars:
+                KUBECONFIG: /var/lib/microshift/resources/kubeadmin/kubeconfig
+            - if:
+              - path: /var/local/acm-import/import.yaml
+                op: [created]
+              run: kubectl apply -f /var/local/acm-import/import.yaml
+              envVars:
+                KUBECONFIG: /var/lib/microshift/resources/kubeadmin/kubeconfig
 ```
 
-Flight Control hooks are explained in the following [link](https://github.com/flightctl/flightctl/blob/main/docs/user/managing-devices.md#using-device-lifecycle-hooks).
-Hooks are triggered at specific moments on the lifecycle of a device such as before/after updating, before/after reboot, etc.
-The hooks definition shown above describe two hooks executed after updating the fleet, one to apply the CRD file and one to apply the rest of the Klusterlet manifests stored in the import.yaml file.
-
-## Device Reconciliation Process
```

Comment:
why drop this? isnt this important information?

Reply:
Because it's way too detailed on implementation for a user documentation and describes either ACM processes documented in the referenced ACM doc or FC processes already described above.